apis/v1alpha1/backendpolicy_types.go

```diff
@@ -0,0 +1,173 @@
+	// given to the oldest BackendPolicy.
+	//
+	// Support: Core
+	//
+	// +optional
+	BackendRefs []BackendRef `json:"backendRefs"`
+
+	// TLS is the TLS configuration for these backends.
+	//
+	// Support: Extended
+	//
+	// +optional
+	TLS *BackendTLSConfig `json:"tls,omitempty"`
+}
+
+// BackendRef identifies an API object within a known namespace that defaults
+// group to core and resource to services if unspecified.
+type BackendRef struct {
+	// Group is the group of the referent.  Omitting the value or specifying
+	// the empty string indicates the core API group.  For example, use the
+	// following to specify a service:
+	//
+	// fooRef:
+	//   resource: services
+	//   name: myservice
+	//
+	// Otherwise, if the core API group is not desired, specify the desired
+	// group:
+	//
+	// fooRef:
+	//   group: acme.io
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +optional
+	// +kubebuilder:default=core
+	Group string `json:"group"`
+
+	// Resource is the API resource name of the referent. Omitting the value
+	// or specifying the empty string indicates the services resource. For example,
+	// use the following to specify a services resource:
+	//
+	// fooRef:
+	//   name: myservice
+	//
+	// Otherwise, if the services resource is not desired, specify the desired
+	// group:
+	//
+	// fooRef:
+	//   group: acme.io
+	//   resource: foos
+	//   name: myfoo
+	//
+	// +optional
+	// +kubebuilder:default=services
+	Resource string `json:"resource"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Port is the port of the referent. If unspecified, this policy applies to
+	// all ports on the backend.
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+}
+
+// BackendTLSConfig describes TLS configuration for a backend.
+type BackendTLSConfig struct {
+	// ClientCertificateRef is a reference to a TLS client certificate that may
+	// be used to connect to these backends. If an entry in this list omits or
+	// specifies the empty string for both the group and the resource, the
+	// resource defaults to "secrets". An implementation may support other
+	// resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
+	//
+	// If a Secret is referenced, it must be of type "kubernetes.io/tls" and
+	// contain tls.crt and tls.key data fields that contain the certificate and
+	// private key to use for TLS.
+	//
+	// Support: Extended
+	//
+	// +optional
+	ClientCertificateRef *CertificateObjectReference `json:"clientCertificateRef,omitempty"`
+
+	// CertificateAuthorityRef is a reference to a resource that includes
+	// trusted CA certificates for the associated backends. If an entry in this
+	// list omits or specifies the empty string for both the group and the
+	// resource, the resource defaults to "secrets". An implementation may
+	// support other resources (for example, resource "mycertificates" in group
+	// "networking.acme.io").
```

Comment:
If we are supporting secrets here, let's add in:
- certificates should be present inside `data["ca.crt"]` key
- they should be PEM encoded
- Multiple certificates can be specified (concat with a newline)

Reply:
Updated, thanks!